File: 617/merge-two-binary-trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} t1
 * @param {TreeNode} t2
 * @return {TreeNode}
 */
var mergeTrees = function (t1, t2) {
    if (!t1) return t2;
    if (!t2) return t1;
    let curr = new TreeNode(t1.val + t2.val);
    curr.left = mergeTrees(t1.left, t2.left);
    curr.right = mergeTrees(t1.right, t2.right);
    return curr;
};

File: 617/merge-two-binary-trees.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func mergeTrees(t1 *TreeNode, t2 *TreeNode) *TreeNode {
	if nil == t1 && nil == t2 {
		return nil
	}
	if nil != t1 && nil == t2 {
		return t1
	} else if nil == t1 && nil != t2 {
		return t2
	} else {
		ret := new(TreeNode)
		ret.Val = t1.Val + t2.Val
		ret.Left = mergeTrees(t1.Left, t2.Left)
		ret.Right = mergeTrees(t1.Right, t2.Right)
		return ret
	}
}